src/compiler/checker.ts

```diff
@@ -745,10 +745,12 @@ namespace ts {
             return _jsxNamespace;
         }
 
-        function getEmitResolver(sourceFile: SourceFile, cancellationToken: CancellationToken) {
+        function getEmitResolver(sourceFile: SourceFile, cancellationToken: CancellationToken, ignoreDiagnostics?: boolean) {
             // Ensure we have all the type information in place for this file so that all the
             // emitter questions of this resolver will return the right information.
-            getDiagnostics(sourceFile, cancellationToken);
+            if (!ignoreDiagnostics) {

```

Comment:
but if we try to emit this file with this emit resolver the emit will fail, e.g. marking aliases as referenced would not have happened.. right?

Reply:
Would not impact declaration file emit right? I mean there could be errors but the output wont change and hence ok to do so?

Reply:
we need to document that well.. also maybe change the parameter name to something more declarative like `declarationFileEmitOnly`. i worry that someone will just create this and pass true by mistake and will be hard to see why it is failing.